src/guides/libraries/_dart-io-tour.md

```diff
@@ -234,27 +234,13 @@ void processRequest(HttpRequest request) {
 
 #### HTTP client
 
-The [HttpClient][] class
-helps you connect to HTTP resources from your Dart command-line or
-server-side application. You can set headers, use HTTP methods, and read
-and write data. The HttpClient class does not work in browser-based
-apps. When programming in the browser, use the
-[dart:html HttpRequest class.][HttpRequest]
-Here's an example of using HttpClient:
-
-<?code-excerpt "misc/test/library_tour/io_test.dart (client)" replace="/Future<\w+\W/void/g"?>
-```dart
-void main() async {
-  var url = Uri.parse('http://localhost:8888/dart');
-  var httpClient = HttpClient();
-  var request = await httpClient.getUrl(url);
-  var response = await request.close();
-  var data = await utf8.decoder.bind(response).toList();
-  print('Response ${response.statusCode}: $data');
-  httpClient.close();
-}
-```
+You should avoid directly using `dart:io` to make HTTP requests.
```

Comment:
Let me know if you think that I should just remove this section completely.

Reply:
I think it's good to keep these sections for two releases or so to point users in the right direction. Curious what Marya thinks though.

In the sections you remove, do consider linking to the `/tutorials/server/fetch-data` page as an alternative to learn more though since these changes remove learning material.

I do think making the corresponding note/warning on the API's dartdocs more prominent will have a larger effect than any changes here.

Reply:
Great suggestion! I've added a link to the tutorial.

There is a code review out for making the corresponding change to the API docs:
https://dart-review.googlesource.com/c/sdk/+/340283

Reply:
We want to be pretty forceful about our recommendation - we are using the same initial wording in the SDK documentation. **But** we are preserving the example so users can find it there.

Wouldn't it be better to avoid having examples in our library tour that should not be used except in rare circumstances?

Reply:
Thanks for explaining, definitely trust your judgement here! 

You can make the final call here, but just to be extra explicit for readers, you could explicitly state that there's still an example in the sdk docs (that's the API docs right?) _if you really need it_ (like Parker said, having an example in one place then making it disappear _too well_ all of a sudden isn't ideal). So, after lines 245 in this file, and 333 at the end of the html-tour file, something like:

```
Though no longer recommended / not recommended for the majority of use cases, 
you can find more information on / an example of `HttpClient`/`HttpRequest` 
in the [API documentation](link-to-api-docs).
```

Approving in the mean time so you can make the final call, but I'll come back and check again if you do change something else and want another look. Thanks for doing this!

Reply:
Isn't the purpose the library tour to indoctrinate developers new to Dart? Those developers should start with `package:http` (or some equivalent package).

I preserved the link to the class and it still has an example there.

Reply:
Oh, and don't trust my judgement; if often sucks ;-)

Reply:
> Isn't the purpose the library tour to indoctrinate developers new to Dart? 

Definitely so. What Parker and I are referring to is the kind of nebulous, hard to measure, off-chance that "what if someone, somewhere has a link to this specific section that they hold onto for reference that says "example here for whatever", and the off-chance that they'll be greatly perturbed if all of a sudden that "whatever"s not there any more and it's not immediately obvious where to find it. 

This definitely isn't a big deal so again, your call :) And your judgement about how important it is that no one remotely even goes poking around the old stuff unless they're absolutely digging for it is based on a lot more context than I have, I'm just citing a general best practice for tech writing. And yes, you do have the link on the API names which I'm ok with!

Reply:
Thanks for the detailed answer. I think that I'm OK with it as-is.